pkg/client/didexchange/client.go

```diff
@@ -304,6 +310,8 @@ func (c *Client) CreateConnection(myDID string, theirDID *did.Doc, options ...Co
 		conn.ServiceEndPoint = destination.ServiceEndpoint
 		conn.RecipientKeys = destination.RecipientKeys
 		conn.RoutingKeys = destination.RoutingKeys
+	} else {
+		return "", fmt.Errorf("create connection: failed to create destination: %w", err)
```

Comment:
Let’s remove `else` condition.
```
if err !=nil {
return "", fmt.Errorf("create connection: failed to create destination: %w", err)
}
```

Reply:
`if err == nil` seems weird

I'll remove the condition altogether. I guess the original author mean if CreateDestination was successful, then set the connection properties in the if clause. Otherwise, create a new connection altogether  (done under this code).

@llorllale @soluchok, should the `else` clause be removed fully or should I add `if err !=nil` as suggested by @soluchok ?

Reply:
should the else clause be removed fully -I don't know that. My suggestion was based on what we have in this PR.

Reply:
I am the author of that part of the code.

The code is allowing the creation of connections with `theirDID` regardless whether it has a valid `did-communication` service or not.

I cannot recall my thinking at the time I wrote this.

Is it valid to create a connection between, say, a `did:peer` and a `did:key`? Or say two `did:peer` where one has `did-communication` and the other does not? The `service` information would have to be transmitted out of band in the latter case presumably.

I think, in the end, we don't have any use case for this right now. Plus I am unable to defend this logic.

I vote we fix it such that `CreateConnection()` fails if `service.CreateDestination()` returns an error.

Reply:
I'll switch the if/else clause to:
```
if err !=nil {
    return "", fmt.Errorf("create connection: failed to create destination: %w", err)
}
conn.ServiceEndPoint = destination.ServiceEndpoint
conn.RecipientKeys = destination.RecipientKeys
conn.RoutingKeys = destination.RoutingKeys
```

Reply:
updated